src/edu/csus/ecs/pc2/shadow/ShadowController.java

```diff
@@ -690,11 +692,33 @@ private Map<String, ShadowJudgementInfo> getJudgementsMap(Run[] pc2Runs) {
                         } else {
                             // we've exhausted methods of obtaining an acronym
                             log.warning("Null judgement acronym for run " + run.getNumber() + ", judgement string " + judgementString + "; skipping");
+
+                            Judgement judgement = localContest.getJudgement(jr.getJudgementId());
+                            if (judgement == null) {
+                                log.warning("Null judgement acronym for run " + run.getNumber() + ", Judgement not found in model for run");
+                            } else {
+
+                                /**
+                                 * Acronym for run judgement
+                                 */
+                                String acronymnName = judgement.getAcronym().toString();
+
+                                // Add ShadowJudgementInfo for run into pc2JudgementInfoMap 
+                                ClientId judgerClientID = jr.getJudgerClientId();
+                                ShadowJudgementPair pair = new ShadowJudgementPair(submissionId, acronymnName, "<pending>");
+                                ShadowJudgementInfo info = new ShadowJudgementInfo(submissionId, teamID, problemID, languageID, judgerClientID, pair);
+                                pc2JudgementInfoMap.put(submissionId, info);
+
+                                log.info("Null judgement acronym for run " + run.getNumber() + ", cjudgement string " + judgementString + "; not skipped assigned acronym = " + acronymnName);
```

Comment:
What does `cjudgement string` mean?  Log messages should make sense to human readers...   Is this just a typo and `cjudgement` was intended to be just `judgement`??